tests/phpunit/tests/user/query.php

```diff
@@ -1725,4 +1728,182 @@ public static function filter_users_pre_query( $posts, $query ) {
 
 		return array( 555 );
 	}
+
+	/**
+	 * @ticket 16841
+	 */
+	public function test_get_single_capability_by_string() {
+		$wp_user_search = new WP_User_Query( array( 'capability' => 'install_plugins' ) );
+		$users          = $wp_user_search->get_results();
+
+		$this->assertNotEmpty( $users );
+		foreach ( $users as $user ) {
+			// User has the capability, but on Multisite they would also need to be a super admin.
+			// Hence using get_role_caps() instead of has_cap().
+			$role_caps = $user->get_role_caps();
+			$this->assertArrayHasKey( 'install_plugins', $role_caps );
+			$this->assertTrue( $role_caps['install_plugins'] );
+		}
+	}
+
+	/**
+	 * @ticket 16841
+	 */
+	public function test_get_single_capability_invalid() {
+		$wp_user_search = new WP_User_Query( array( 'capability' => 'foo_bar' ) );
+		$users          = $wp_user_search->get_results();
+
+		$this->assertEmpty( $users );
+	}
+
+	/**
+	 * @ticket 16841
+	 */
+	public function test_get_single_capability_by_array() {
+		$wp_user_search = new WP_User_Query( array( 'capability' => array( 'install_plugins' ) ) );
+		$users          = $wp_user_search->get_results();
+
+		$this->assertNotEmpty( $users );
+		foreach ( $users as $user ) {
+			// User has the capability, but on Multisite they would also need to be a super admin.
+			// Hence using get_role_caps() instead of has_cap().
+			$role_caps = $user->get_role_caps();
+			$this->assertArrayHasKey( 'install_plugins', $role_caps );
+			$this->assertTrue( $role_caps['install_plugins'] );
+		}
+	}
+
+	/**
+	 * @ticket 16841
+	 */
+	public function test_get_single_capability_added_to_user() {
+		$user = self::factory()->user->create_and_get( array( 'role' => 'subscriber' ) );
+		$user->add_cap( 'custom_cap' );
```

Comment:
If the global `tearDown()` resets roles, are you able to use `self::$subscriber` instead?

:1234: for admin in `test_get_multiple_capabilities_should_only_match_users_who_have_each_capability_added_to_user()`